### ISSUE TITLE
Christoffel Symbols are no tensor!

### ISSUE THREAD

**Issue by andi8086** [einsteinpy/einsteinpy#497]
🐞 Christoffel-Symbols are treated like tensors by API

For formatting or internal handling or whatever, we have
ch = ChristoffelSymbols.from_metric(m_obj)
ch.tensor()

🎯 Maybe better say Matrix instead of Tensor in this case

Tensors are multilinear maps. Just because something can be written
as a matrix does not mean it is a tensor.

It is very dangerous to refer to quantities as tensor if they are not, i.e.
follow a different coordinate transformation rule...

A tensor is a geometric object, i.e. if null it is null in all coordinate systems. The
christoffel symbols however can espacially transformed away in a free falling
reference frame. That would never be possible with a tensor.

This can cause a lot of confusion especially for students that use this software.

**Comment by ritzvik** [comment]
@andi8086 Thank you very much for pointing this out. We know about the problem which as discussed in #374 . We'll surely think of a workaround.

**Comment by WenyinWei** [comment]
The `self.matrix()` suggestion still failed to match Christoffel symbols, how about `self.array()`

**Comment by WenyinWei** [comment]
The same argument also applies to the Levi-Civita symbol (not yet exist in einsteinpy), which are not tensors indeed but can `tensor_product` and `tensor_cross` with other tensors in the way like a tensor. However, they do not follow the tensor way of config raising or lowering. So whether or not to inherit a new class `SpecialConfigChangeArray` from `BaseRelativityTensor` would need to taken carefully, since these Levi-Civita and Christoffel (Array) symbols are not tensors in fact. I prefer to call `BaseRelativityTensor` as `Tensor` while `einsteinpy.symbolic.tensor.Tensor` as `PseudoTensor` or `QuasiTensor`. `einsteinpy.symbolic.tensor.Tensor` does not offer the `change_config` method, so it fits the requirement for such non-tensor arrays.

**Comment by JeS24** [comment]
@WenyinWei Right. `array()` is an obvious choice here. Also, this is how the current `tensor()` attribute is defined:
```py
def tensor(self):
    """
    Returns the sympy Array

    Returns
    -------
    ~sympy.tensor.array.dense_ndim_array.ImmutableDenseNDimArray
        Sympy Array object

    """
    return self.arr
```
We are already returning "array" to the user, when `tensor()` is invoked. It would not take much work to edit this method to return `array()` instead. We can even make this a `@property`. We'd need a `DeprecationWarning` here though. Labeling this as a `good first issue`.

Also, for general tensor / non-tensor ops, it is perhaps better to have separate classes or mixins that define the operations, that appropriate objects can then derive from. This would avoid the issue of conflating dissimilar mathematical objects. We have a [`TransformationMatrix`](https://github.com/einsteinpy/einsteinpy/blob/3926ea7f0d6619adc6e53f7cb34170c41fb17b35/src/einsteinpy/symbolic/helpers.py#L67) class, that can be refined or extended.

**Comment by Whitris** [comment]
Is this issue still open? May I work on it?
